src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeName.cs

```diff
@@ -69,6 +69,25 @@ sealed class TypeName
 #endif
         }
 
+#if SYSTEM_REFLECTION_METADATA
+        private TypeName(string? fullName,
+            AssemblyNameInfo? assemblyName,
+            TypeName? elementOrGenericType = default,
+            TypeName? declaringType = default,
+            ImmutableArray<TypeName> genericTypeArguments = default,
```

Comment:
```suggestion
            TypeName? elementOrGenericType,
            TypeName? declaringType,
            ImmutableArray<TypeName> genericTypeArguments,
```
These default values do not seem to be useful. The code specifies them explicitly in nearly all cases.